### PR TITLE
Docs 1.0

### PR DESCRIPTION
**Pull request by LewisJEllis**
I probably forgot something somewhere, but I think these are done enough to publish.

/cc @benvinegar @MaxBittker @mattrobenolt

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -1,17 +1,28 @@
 Configuration
 =============
 
-Configuration is passed as the second argument of the ``raven.Client`` constructor:
+To get started, you need to configure Raven.js to use your Sentry DSN:
```
Comment (benvinegar) [inline docs/config.rst]: > Raven.js

Intentional?

File: docs/config.rst [hunk 2]
```diff
@@ -81,6 +92,40 @@ Optional settings
...
+      autoBreadcrumbs: {
+          'console': false,  // console logging
+          'http': true,     // http and https requests
+          'postgres': true,  // postgresql queries from pg module
+      }
+
+.. describe:: maxBreadcrumbs
+
+  Raven captures up to 30 breadcrumb entries by default. You can increase this to
+  be as high as 100, or reduce it if you find 30 is too noisy, by setting `maxBreadcrumbs`.
```
Comment (benvinegar) [inline docs/config.rst]: I think this needs a note about potential memory consumption.

File: docs/index.rst [hunk 3]
```diff
@@ -10,7 +10,7 @@
     Node.js
     =======
 
-raven-node is a Sentry's officially supported Node.js SDK.
+raven-node is the officially supported Node.js client for Sentry.
```
Comment (benvinegar) [inline docs/index.rst]: This can be more concise:

> is the official Node.js client for Sentry.

File: docs/index.rst [hunk 4]
```diff
@@ -32,55 +32,89 @@ Next you need to initialize the Raven client and configure it to use your `Sentr
...
+  });
+  // errors thrown here will not be associated with matt
 
-If at any point, the user becomes unauthenticated, you can call
-``client.setUserContext()`` with no arguments to remove their data.
+This can also be used to set ``tags`` and ``extra`` keys for associated tags and extra data.
 
-Other similar methods are ``client.setExtraContext`` and
-``client.setTagsContext``.  See :ref:`raven-node-additional-context` for more info.
+We can update our context data with ``mergeContext`` or retrieve it with ``getContext``. When an exception is captured by a wrapper, the current context state will be passed as options to ``captureException``.
```
Comment (benvinegar) [inline docs/index.rst]: Let's not use "we" here – it should be "you" and "your". There are other places where this is inconsistent (e.g. the CORS paragraph in raven-js docs), and we should fix those too.

File: docs/integrations/express.rst [hunk 5]
```diff
@@ -4,26 +4,27 @@ Express
...
-        res.statusCode = 500;
-        res.end(res.sentry+'\n');
-    }
+    // Must configure Raven before doing anything else with it
+    Raven.config('__DSN__').install();
 
-    // The request handler must be the first item
-    app.use(raven.middleware.express.requestHandler('___DSN___'));
+    // The request handler must be the first middleware on the app
+    app.use(Raven.requestHandler());
```
Comment (benvinegar) [inline docs/integrations/express.rst]: So much better ...

File: docs/integrations/express.rst [hunk 6]
```diff
@@ -4,26 +4,27 @@ Express
...
-    app.use(raven.middleware.express.errorHandler('___DSN___'));
+    app.use(Raven.errorHandler());
 
     // Optional fallthrough error handler
-    app.use(onError);
+    app.use(function onError(err, req, res, next) {
+        // The error id is attached to `res.sentry` to be returned
+        // and optionally displayed to the user for support.
+        res.statusCode = 500;
+        res.end(res.sentry+'\n');
```
Comment (benvinegar) [inline docs/integrations/express.rst]: Add space between `res.sentry` and `\n` like you'd done above?

File: docs/usage.rst [hunk 7]
```diff
@@ -4,43 +4,136 @@ Usage
...
+  var Raven = require('raven');
+
+  Raven.config('___DSN___').install();
+  Raven.context(function () {
+    // all your stuff goes here
+  });
+
+Raven will automatically catch and report any unhandled errors originating inside this function
+(or anything it calls, etc), so you don't have to manually `captureException` everywhere. This
+also gives your code access to the context methods. See below for more on contexts.
```
Comment (benvinegar) [inline docs/usage.rst]: > also gives your code access to the context methods

"to context methods"? (drop "the")?

File: docs/usage.rst [hunk 8]
```diff
@@ -4,43 +4,136 @@ Usage
...
+.. code-block:: javascript
+
+    var doIt = function () {
+        // doing cool stuff
     }
 
+    setTimeout(Raven.wrap(doIt), 1000)
+
+We refer to code wrapped via ``Raven.context`` or ``Raven.wrap`` as being inside a context. Code inside a context
+has access to the ``getContext``, ``mergeContext``, and ``getContext`` methods for associating data with that context.
```
Comment (benvinegar) [inline docs/usage.rst]: `getContext` is listed twice

File: docs/usage.rst [hunk 9]
```diff
@@ -198,4 +318,32 @@ Passing any falsey value as the DSN will disable sending events upstream:
...
+These alerts are hopefully helpful during initial setup or in upgrading Raven versions,
+but once you have everything set up and going, we recommend disabling them:
+
+.. code-block:: javascript
+
+  Raven.disableConsoleAlerts();
+
+Multiple Instances
+------------------
+Normally we have just one instance of Raven:
```
Comment (benvinegar) [inline docs/usage.rst]: > Normally there is just one

**Comment by LewisJEllis** [comment]
@benvinegar pushed a commit that should address all these.

**Pull request merged**